src/ProductView/TabBar/QueryInterface/QueryInterface.js

```diff
@@ -118,10 +121,23 @@ class QueryInterface extends Component {
     }
 
     handleChangeQuery(event) {
-        this.setState({ eventQuery: event.target.value });
+        const queryValue = event.target.value;
+        const modalIsAbleToSave = this.isModalAbleToSave(this.state.eventTypeName, queryValue);
+        this.setState({ eventQuery: queryValue, modalIsAbleToSave: modalIsAbleToSave });
+        this.eventQueryFormValidation(queryValue);
+    }
+
+    isModalAbleToSave(eventTypeName, eventQuery) {
+        if (eventTypeName && eventQuery) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1126)